src/xrpld/app/tx/detail/LoanBrokerDelete.cpp

```diff
@@ -0,0 +1,197 @@
+#include <xrpld/app/tx/detail/LoanBrokerDelete.h>
+//
+#include <xrpld/app/misc/LendingHelpers.h>
+
+namespace ripple {
+
+bool
+LoanBrokerDelete::checkExtraFeatures(PreflightContext const& ctx)
+{
+    return checkLendingProtocolDependencies(ctx);
+}
+
+NotTEC
+LoanBrokerDelete::preflight(PreflightContext const& ctx)
+{
+    if (ctx.tx[sfLoanBrokerID] == beast::zero)
+        return temINVALID;
+
+    return tesSUCCESS;
+}
+
+TER
+LoanBrokerDelete::preclaim(PreclaimContext const& ctx)
+{
+    auto const& tx = ctx.tx;
+
+    auto const account = tx[sfAccount];
+    auto const brokerID = tx[sfLoanBrokerID];
+
+    auto const sleBroker = ctx.view.read(keylet::loanbroker(brokerID));
+    if (!sleBroker)
+    {
+        JLOG(ctx.j.warn()) << "LoanBroker does not exist.";
+        return tecNO_ENTRY;
+    }
+
+    auto const brokerOwner = sleBroker->at(sfOwner);
+
+    if (account != brokerOwner)
+    {
+        JLOG(ctx.j.warn()) << "Account is not the owner of the LoanBroker.";
+        return tecNO_PERMISSION;
+    }
+    if (auto const ownerCount = sleBroker->at(sfOwnerCount); ownerCount != 0)
+    {
+        JLOG(ctx.j.warn()) << "LoanBrokerDelete: Owner count is " << ownerCount;
+        return tecHAS_OBLIGATIONS;
+    }
+    if (auto const debtTotal = sleBroker->at(sfDebtTotal);
+        debtTotal != beast::zero)
+    {
+        // Any remaining debt should have been wiped out by the last Loan
+        // Delete. This check is purely defensive.
+        auto const vault =
+            ctx.view.read(keylet::vault(sleBroker->at(sfVaultID)));
+        if (!vault)
+            return tefINTERNAL;  // LCOV_EXCL_LINE
+        auto const asset = vault->at(sfAsset);
+        auto const scale = getVaultScale(vault);
+
+        auto const rounded =
+            roundToAsset(asset, debtTotal, scale, Number::towards_zero);
+
+        if (rounded != beast::zero)
+        {
+            // LCOV_EXCL_START
+            JLOG(ctx.j.warn()) << "LoanBrokerDelete: Debt total is "
+                               << debtTotal << ", which rounds to " << rounded;
+            return tecHAS_OBLIGATIONS;
+            // LCOV_EXCL_START
+        }
+    }
+
+    auto const vault = ctx.view.read(keylet::vault(sleBroker->at(sfVaultID)));
+    if (!vault)
+    {
+        // LCOV_EXCL_START
+        JLOG(ctx.j.fatal()) << "Vault is missing for Broker " << brokerID;
+        return tefBAD_LEDGER;
```

Comment:
Why is the error code different from  another vault check above at line 57? Shouldn't it be the same? Also can optimize and check non-null vault once before checking non-zero debtTotal.

Reply:
> Why is the error code different from another vault check above at line 57? Shouldn't it be the same? Also can optimize and check non-null vault once before checking non-zero debtTotal.

Because two different people were working in this same general area at the same time, and nobody noticed the discrepancy when merging.

Fix is pending in #6161.